### PR TITLE
fix: correct notification preference type

### PR DESCRIPTION
**Pull request by sshanzel**
after further development, I noticed that even though we strictly set the proper `type` of the `NotificationPreference`, it is getting overridden by what was defined in the class. Using the base class results in always setting it to `NotificationPreference` string as shown below. I have pushed a commit to fix this and ended up with a switch case, I tried doing a map of the class but wasn't working.

Screenshot 2023-07-26 at 7 24 51 PM

### CODE REVIEW

**Review by capJavert — APPROVED** [review]
Looks like it's a running issue from even before in our repo and typeorm:
- https://github.com/dailydotdev/daily-api/pull/1154
- https://github.com/typeorm/typeorm/issues/9033

Looks ok from my side 👍

**Review by idoshamun — COMMENTED** [review]
We should add tests to check the right type then. To make sure we don't introduce it again

File: __tests__/notifications.ts [hunk 1]
```diff
@@ -551,6 +552,9 @@ describe('mutation muteNotificationPreference', () => {
       .findOneBy(params);
 
     expect(muted).toBeTruthy();
+    expect(muted.type).toEqual(
+      notificationPreferenceMap[params.notificationType],
+    );
```
Comment (capJavert) [inline __tests__/notifications.ts:557]: This test I think covers the type check during insert 👍

Comment (sshanzel) [reply __tests__/notifications.ts:557]: Yes, this is the one I also used to make the test fail, and applied the fix to make this go green.

Comment (sshanzel) [reply __tests__/notifications.ts:557]: cc @idoshamun

Comment (idoshamun) [reply __tests__/notifications.ts:557]: Missed that!

**Review by idoshamun — APPROVED** [review]

**Pull request merged**